### PR TITLE
ENG-527: Support mintable erc 1155

### PR DESCRIPTION
**Pull request by WilsonSilvaThreeSigma**
## Describe your changes

Add support for mintable ERC-1155.
Added encoding message according to this definitions:
- [Mintable ERC-1155](https://docs.starkware.co/starkex/spot/shared/starkex-specific-concepts.html#mintable_erc_1155)
- [Assigned minting request bits per asset type](https://docs.starkware.co/starkex/spot/offchain-minting.html#starkex_validates_mint_request)

## Issue ticket number and link

[ENG-527](https://linear.app/threesigma/issue/ENG-527/support-mintable-erc1155-and-asset-encoding-for-mint-amounts-1in)

## Checklist before requesting a review
- [X] I have performed a self-review of my code
- [X] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.

## Reviewers
@JoseAfonsoThreeSigma @asynctomatic

### CODE REVIEW

File: src/StarkEx.Crypto.SDK/Encoding/AssetEncoder.cs [hunk 1]
```diff
@@ -202,6 +218,53 @@ private static string GetMintableErc20AssetId(string mintingBlob, string address
...
+            .ClearBit(241)
+            .ClearBit(242)
+            .ClearBit(243)
+            .ClearBit(244)
+            .ClearBit(245)
+            .ClearBit(246)
+            .ClearBit(247)
+            .SetBit(248)
+            .SetBit(249)
+            .SetBit(250);
```
Comment (JoseAfonsoThreeSigma) [inline src/StarkEx.Crypto.SDK/Encoding/AssetEncoder.cs:232]: I'm unsure if these new encodings should always be applied or should only be applied when mint amounts are bigger than one. Will ask StarkEx team about this

File: tests/StarkEx.Crypto.SDK.Tests/Encoding/AssetEncoderTests.cs [hunk 2]
```diff
@@ -16,8 +16,9 @@ public class AssetEncoderTests
...
     [InlineData(AssetType.Erc721, "1", "0xB18ed4768F87b0fFAb83408014f1caF066b91380", "1004", null, "0x2b0ff0c09505bc40f9d1659becf16855a7b2298b010f8a54f4b05325885b40c")]
     [InlineData(AssetType.Erc1155, "1", "0x22c36BfdCef207F9c0CC941936eff94D4246d14A", "0x539", null, "0x3bac60418017ad6c32f23980201722fbe672d9bd108765469484347b00afda")]
-    [InlineData(AssetType.MintableErc20, "1000000000", "0x5da41d8b03b656ac0daac9f27b98feba461dfbad", null, "0xdeadbeef", "0x400f163c4d559288a2edbb10162eed11f4de87c56875b970fee1534da69cc80")]
+    [InlineData(AssetType.MintableErc20, "1000000000", "0x5da41d8b03b656ac0daac9f27b98feba461dfbad", null, "0xdeadbeef", "0x700f163c4d559288a2edbb10162eed11f4de87c56875b970fee1534da69cc80")]
     [InlineData(AssetType.MintableErc721, "1", "0x7a42586b2ab661458097094582b22b3c05342bd9", null, "0xdeadbeef", "0x400715adb86fb84bc0c27f4aff1f78fb6118b5b5e978fd7e0e047634a3a56d9")]
+    [InlineData(AssetType.MintableErc1155, "1", "0x7a42586b2ab661458097094582b22b3c05342bd9", null, "0xdeadbeef", "0x60047b06b775209daf643439760dd879b53d0409b50a5eb02b87bce1aa7818a")]
```
Comment (JoseAfonsoThreeSigma) [inline tests/StarkEx.Crypto.SDK.Tests/Encoding/AssetEncoderTests.cs:21]: Did you get these results from somewhere? Or are these the outputs generated by our own code? Ideally we would compare our results against an official implementation, which sadly doesn't exist yet

Comment (WilsonSilvaThreeSigma) [reply tests/StarkEx.Crypto.SDK.Tests/Encoding/AssetEncoderTests.cs:21]: Unfornately, they have docs on the implementation but no output values, so it was only outputs generated by our own code

File: src/StarkEx.Crypto.SDK/Encoding/AssetEncoder.cs (hunk 1 above)
Comment (WilsonSilvaThreeSigma) [reply src/StarkEx.Crypto.SDK/Encoding/AssetEncoder.cs:232]: It's not clear on the docs, thanks for asking them!

**Review by 3s-jorgematos — APPROVED** [review]
LGTM

**Review by 3s-jorgematos — APPROVED** [review]

**Review by JoseAfonsoThreeSigma — COMMENTED** [review]
Can we have the param `isMintingTransaction` with a default value of `false`?

**Comment by WilsonSilvaThreeSigma** [comment]
> Can we have the param `isMintingTransaction` with a default value of `false`?

We already do 
```
public static string GetAssetId(
        AssetType assetType,
        string mintingBlob = null,
        string address = null,
        string tokenId = null,
        BigInteger quantum = null)
        BigInteger quantum = null,
        bool isMintingTransaction = false)
```

**Review by JoseAfonsoThreeSigma — APPROVED** [review]

**Pull request merged**